src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -133,22 +132,14 @@ void SpatialPooler::setGlobalInhibition(bool globalInhibition) {
   globalInhibition_ = globalInhibition;
 }
 
-Int SpatialPooler::getNumActiveColumnsPerInhArea() const {
-  return numActiveColumnsPerInhArea_;
-}
-
-void SpatialPooler::setNumActiveColumnsPerInhArea(UInt numActiveColumnsPerInhArea) {
-  NTA_CHECK(numActiveColumnsPerInhArea > 0u && numActiveColumnsPerInhArea <= numColumns_); //TODO this boundary could be smarter
-  numActiveColumnsPerInhArea_ = numActiveColumnsPerInhArea;
-  localAreaDensity_ = DISABLED;  //MUTEX with localAreaDensity
-}
 
 Real SpatialPooler::getLocalAreaDensity() const { return localAreaDensity_; }
 
-void SpatialPooler::setLocalAreaDensity(Real localAreaDensity) {
+void SpatialPooler::setLocalAreaDensity(const Real localAreaDensity) {
   NTA_CHECK(localAreaDensity > 0.0f && localAreaDensity <= 1.0f);
+  NTA_CHECK(static_cast<UInt>(localAreaDensity * getNumColumns()) > 0) 
+	  << "Too small density or sp.getNumColumns() -> would have zero active output columns.";
```

Comment:
This is an important check that caused many SP tests to "break". Ensure sparsity & num columns are such that there will be some >0 desired active columns. (with numActivePerInhArea this was a different issue, for sparsity it had been broken)

- [ ] TODO followup: bump this number further up, say 40, to ensure some stability and noise robustness of the output SDRs (SDR with 1 active bit is not really good). 
   - [ ] I can make this a constant in SP
   - [ ] but actually bumping it would again break many tests that use too trivial, non-realisting settings to demonstrate just the one idea being tested.